pkg/client/events/replay_client_integration_test.go

```diff
@@ -168,15 +156,15 @@ func TestReplayClient_Remapping(t *testing.T) {
 		}
 	}()
 
-	// Wait for ~2 events to be received
-	time.Sleep(51600 * time.Microsecond)
-	// Close the connection
+	// Wait for ~2 events to be received before closing the connection
```

Comment:
Why is this the amount of time we're waiting for 2 events? It's always a bit of a code smell when tests are this dependent on specific times (instead of being event driven), so just looking to understand and #PUC.